app/contracts/base_contract.rb

```diff
@@ -250,7 +247,8 @@ def reduce_writable_attributes(attributes)
 
   def reduce_by_writable_conditions(attributes)
     collect_ancestor_attributes(:writable_conditions).each do |attribute, condition|
-      attributes -= [attribute, "#{attribute}_id"] unless instance_exec(&condition)
+      condition = !!instance_exec(&condition) if condition.respond_to?(:call)
+      attributes -= [attribute, "#{attribute}_id"] if condition == false
```

Comment:
To make sure I understand correctly, the behavior should now be:

- if the condition is a lambda, execute it to test if it resolves to `true`/`false`
- if it isn't, keep the condition that might be coming in as a primitive `true`/`false` boolean value
- use this condition to determine whether to remove the attribute from the list of writable ones

?

Reply:
Yes, exactly